src/properties.ts

```diff
@@ -289,39 +289,61 @@ function getBootstrapperProperties(startTimestampMs: number): ScannerProperties
 /**
  * Get endpoint properties from scanner properties.
  */
-export function getHostProperties(properties: ScannerProperties): ScannerProperties {
-  let sonarHostUrl = properties[ScannerProperty.SonarHostUrl] ?? '';
-
-  if (!sonarHostUrl || SONARCLOUD_URL_REGEX.exec(sonarHostUrl)) {
+export function getHostProperties(serverUrl?: string, sonarCloudUrl?: string): ScannerProperties {
```

Comment:
what was the rational to move these outside of this method?
```
const sonarHostUrl = properties[ScannerProperty.SonarHostUrl];
const sonarCloudUrl = properties[ScannerProperty.SonarScannerSonarCloudURL];
const hostProperties = getHostProperties(sonarHostUrl, sonarCloudUrl);
```

I might be missing something, but I don't see them used anywhere else

Reply:
Basically, the idea was to make it clear when reading the code in `getProperties(..)` that this function is "simple" and only expects the server url and SonarCloud URL

```typescript
  const sonarHostUrl = properties[ScannerProperty.SonarHostUrl];
  const sonarCloudUrl = properties[ScannerProperty.SonarScannerSonarCloudURL];
  const hostProperties = getHostProperties(sonarHostUrl, sonarCloudUrl);
// vs
  const hostProperties = getHostProperties(properties);
```

Taking a second look at this and given your doubts, this choice is probably questionable in that it also adds a bit of complexity to the code